read-fonts/src/tables/gsub/closure.rs

```diff
@@ -579,6 +582,211 @@ fn intersect_coverage(
     Some(r).filter(|set| !set.is_empty())
 }
 
+impl SubstitutionLookupList<'_> {
+    pub fn closure_lookups(
+        &self,
+        glyph_set: &IntSet<GlyphId>,
+        lookup_indices: &mut IntSet<u16>,
+    ) -> Result<(), ReadError> {
+        let mut c = LookupClosureCtx::new(glyph_set);
+
+        let lookups = self.lookups();
+        for idx in lookup_indices.iter() {
+            let lookup = lookups.get(idx as usize)?;
+            lookup.closure_lookups(&mut c, idx)?;
+        }
+
+        lookup_indices.union(c.visited_lookups());
+        lookup_indices.subtract(c.inactive_lookups());
+        Ok(())
+    }
+}
+
+impl LookupClosure for SubstitutionLookup<'_> {
+    fn closure_lookups(
+        &self,
+        c: &mut LookupClosureCtx,
+        lookup_index: u16,
+    ) -> Result<(), ReadError> {
+        if c.lookup_visited(lookup_index) {
+            return Ok(());
+        }
+
+        c.set_lookup_visited(lookup_index);
+        if !self.intersects(c.glyphs())? {
+            c.set_lookup_inactive(lookup_index);
+            return Ok(());
+        }
+
+        let lookup_type = self.lookup_type();
+        self.subtables()?.closure_lookups(c, lookup_type)
+    }
+
+    fn intersects(&self, glyph_set: &IntSet<GlyphId>) -> Result<bool, ReadError> {
+        self.subtables()?.intersects(glyph_set)
+    }
+}
+
+impl LookupClosure for SubstitutionSubtables<'_> {
+    fn closure_lookups(&self, _c: &mut LookupClosureCtx, _arg: u16) -> Result<(), ReadError> {
+        Ok(())
+    }
```

Comment:
not necessary right? because of the default impl?

Reply:
oh I'll implement it in later PRs